projects/js-packages/charts/src/components/legend/base-legend.tsx

```diff
@@ -1,79 +1,107 @@
-import { LegendOrdinal } from '@visx/legend';
+import { LegendItem, LegendLabel, LegendOrdinal, LegendShape } from '@visx/legend';
 import { scaleOrdinal } from '@visx/scale';
 import clsx from 'clsx';
+import { useCallback, type FC } from 'react';
 import styles from './legend.module.scss';
+import { valueOrIdentity, valueOrIdentityString, labelTransformFactory } from './utils';
 import type { LegendProps } from './types';
-import type { FC } from 'react';
 
-/**
- * Base legend component that displays color-coded items with labels using visx
- * @param {object} props             - Component properties
- * @param {Array}  props.items       - Array of legend items to display
- * @param {string} props.className   - Additional CSS class names
- * @param {string} props.orientation - Layout orientation (horizontal/vertical)
- * @return {JSX.Element}               Rendered legend component
- */
 const orientationToFlexDirection = {
 	horizontal: 'row' as const,
 	vertical: 'column' as const,
 };
 
+/*
+ * Base legend component that displays color-coded items with labels based on visx LegendOrdinal.
+ * We avoid using LegendOrdinal directly to enable support for advanced features such as interactivity.
```

Comment:
This component is almost the same as the [visx's Legend](https://github.com/airbnb/visx/blob/master/packages/visx-legend/src/legends/Legend/index.tsx) component, but with a few customizations. In the future, we could improve our legend to support interactive features by passing event handlers to legend items.